test/behaviors/conference.behavior.js

```diff
@@ -533,6 +532,121 @@ function shouldBehaveLikeConference () {
+
+      assertBalanceWithDeposit((await getBalance(conference.address)), 0)
+    })
+
+    it('calling with a number of attenders', async function() {
+      let previousBalanceOne = await getBalance(accounts[11]);
+      let previousBalanceTwo = await getBalance(accounts[12]);
+
+      await wait(20, 1);
+      await conference.clearAndSend(1);
+      
+      let diffOne = new EthVal(await getBalance(accounts[11])).sub(previousBalanceOne)
+      let diffTwo = new EthVal(await getBalance(accounts[12])).sub(previousBalanceTwo)
+      
+      assert.isOk(diffOne.eq(payoutAmount.sub(fees)))
+      assert.isOk(diffTwo.lt(payoutAmount.sub(fees)))
+
+      assertBalanceWithDeposit((await getBalance(conference.address)), mulBN(payoutAmount, 1))
+    })
+
+    it('calling with a number of attendees greater than the contract\'s one', async function() {
+      await wait(20, 1);
+      await conference.clearAndSend(5).should.be.rejected;
+
+      assertBalanceWithDeposit((await getBalance(conference.address)), mulBN(payoutAmount, 2))
+    })
+
+    it('consecutive calls', async function() {
+      let previousBalanceOne = await getBalance(accounts[11]);
+      let previousBalanceTwo = await getBalance(accounts[12]);
+
+      await wait(20, 1);
+      await conference.clearAndSend(1);
+      await conference.clearAndSend(1);
+      
+      let diffOne = new EthVal(await getBalance(accounts[11])).sub(previousBalanceOne)
+      let diffTwo = new EthVal(await getBalance(accounts[12])).sub(previousBalanceTwo)
+      
+      assert.isOk(diffOne.eq(payoutAmount.sub(fees)))
+      assert.isOk(diffTwo.eq(payoutAmount.sub(fees)))
+
+      assertBalanceWithDeposit((await getBalance(conference.address)), mulBN(payoutAmount, 0))
+    })
+
+    it('paid users are not refunded', async function() {
+      await wait(20, 1);
+
+      await conference.withdraw({from:accounts[11]});
+
+      let previousBalanceOne = await getBalance(accounts[11]);
+      let previousBalanceTwo = await getBalance(accounts[12]);
+      await conference.clearAndSend();
+
+      let diffOne = new EthVal(await getBalance(accounts[11])).sub(previousBalanceOne)
+      let diffTwo = new EthVal(await getBalance(accounts[12])).sub(previousBalanceTwo)
+      
+      assert.isOk(diffOne.eq(0))
+      assert.isOk(diffTwo.eq(payoutAmount.sub(fees)))
+
+      assertBalanceWithDeposit((await getBalance(conference.address)), mulBN(payoutAmount, 0))
+    })
+
+    it('users can\'t withdraw after clear and send is called', async function() {
+      await wait(20, 1);
+      let previousBalanceOne = await getBalance(accounts[11]);
+      await conference.clearAndSend();
+      await conference.withdraw({from:accounts[11]}).should.be.rejected;
+      let diffOne = new EthVal(await getBalance(accounts[11])).sub(previousBalanceOne)
+      assert.isOk(diffOne.lte(payoutAmount.sub(fees))) // for eth: due to gas
+      assertBalanceWithDeposit((await getBalance(conference.address)), mulBN(payoutAmount, 0))
+    })
+
```

Comment:
Can you add the following case to make sure that a user can withdraw even after clearAndSend is called as long as the user is not withrawn nor we send money back?

- clearAndSend(1)
- conference.withdraw({from:accounts[11]}).should.be.rejected;
- conference.withdraw({from:accounts[12]}) // this should still pass

Reply:
When you write the test, I suggest you call `conference. isPaid(address)` to make sure which one is paid and which one is not paid after calling `clearAndSend`